logstash-core-event/lib/logstash/string_interpolation.rb

```diff
@@ -124,9 +124,11 @@ def evaluate(event)
         value.join(",")
       when Hash
         LogStash::Json.dump(value)
+      when Numeric
+        value
       else
-        # Make sure we dont work on the refence of the value
-        # The Java Event implementation was always returning a string.
+        # If we dont know how to deal with the type we return a string
+        # to make sure we don't return a reference.
```

Comment:
If you want to be full backward compatible yes, we should we didn't have any test before covering the boolean cases.